core/src/ecmult.rs

```diff
@@ -9,18 +13,149 @@ pub const ECMULT_TABLE_SIZE_A: usize = 1 << (WINDOW_A - 2);
+    }
+
+    /// Deconstruct the `ECMultContext` into raw `AffineStorage`.
+    pub fn deconstruct_raw(self) -> [AffineStorage; ECMULT_TABLE_SIZE_G] {
+        self.pre_g
+    }
+
+    /// Generate a new `ECMultContext` in the heap. Note that this function is expensive.
+    pub fn new_boxed() -> Box<Self> {
+        let mut this = unsafe {
+            let ptr = alloc(Layout::new::<ECMultContext>()) as *mut ECMultContext;
+            let mut this = Box::from_raw(ptr);
+
+            for i in 0..ECMULT_TABLE_SIZE_G {
+                this.pre_g[i] = AffineStorage::default();
+            }
+
+            this
+        };
+
+        let mut gj = Jacobian::default();
+        gj.set_ge(&AFFINE_G);
+        odd_multiples_table_storage_var(&mut this.pre_g, &gj);
+
+        this
+    }
 }
 
+pub fn set_all_gej_var(a: &[Jacobian]) -> Vec<Affine> {
+    let mut az: Vec<Field> = Vec::with_capacity(a.len());
+    for i in 0..a.len() {
+        if !a[i].is_infinity() {
+            az.push(a[i].z.clone());
+        }
+    }
+    let azi: Vec<Field> = inv_all_var(&az);
+
+    let mut ret = Vec::with_capacity(a.len());
+    for _ in 0..a.len() {
+        ret.push(Affine::default());
+    }
+
+    let mut count = 0;
+    for i in 0..a.len() {
+        ret[i].infinity = a[i].infinity;
+        if !a[i].is_infinity() {
+            ret[i].set_gej_zinv(&a[i], &azi[count]);
+            count += 1;
+        }
+    }
+    ret
+}
+
+/// Calculate the (modular) inverses of a batch of field
+/// elements. Requires the inputs' magnitudes to be at most 8. The
+/// output magnitudes are 1 (but not guaranteed to be
+/// normalized). The inputs and outputs must not overlap in
+/// memory.
+pub fn inv_all_var(fields: &[Field]) -> Vec<Field> {
+    if fields.len() == 0 {
+        return Vec::new();
+    }
+
+    let mut ret = Vec::new();
+    ret.push(fields[0].clone());
+
+    for i in 1..fields.len() {
+        ret.push(Field::default());
+        ret[i] = &ret[i - 1] * &fields[i];
+    }
+
+    let mut u = ret[fields.len() - 1].inv_var();
+
+    for i in (1..fields.len()).rev() {
+        let j = i;
+        let i = i - 1;
+        ret[j] = &ret[i] * &u;
+        u = &u * &fields[j];
+    }
+
+    ret[0] = u;
+    ret
+}
+
+const GEN_BLIND: Scalar = Scalar(
+    [2217680822, 850875797, 1046150361, 1330484644,
+     4015777837, 2466086288, 2052467175, 2084507480]
+);
+const GEN_INITIAL: Jacobian = Jacobian {
+    x: Field::new_raw(
+        586608, 43357028, 207667908, 262670128, 142222828,
+        38529388, 267186148, 45417712, 115291924, 13447464
+    ),
+    y: Field::new_raw(
+        12696548, 208302564, 112025180, 191752716, 143238548,
+        145482948, 228906000, 69755164, 243572800, 210897016
+    ),
+    z: Field::new_raw(
+        3685368, 75404844, 20246216, 5748944, 73206666,
+        107661790, 110806176, 73488774, 5707384, 104448710
+    ),
+    infinity: false,
+};
+
 /// Context for accelerating the computation of a*G.
 pub struct ECMultGenContext {
```

Comment:
Maybe dumb q but does this have to be `pub`?

Reply:
Library users need it when they use dynamic context. Something like this:

```
let context: Box<ECMultGenContext> = ECMultGenContext::new_boxed();
let public = PublicKey::from_secret_key_with_context(&secret_key, &context);
```